pdfbox/src/main/java/org/apache/pdfbox/pdmodel/common/function/PDFunction.java

```diff
@@ -348,6 +348,9 @@ else if (x > rangeMax)
      */
     protected float interpolate(float x, float xRangeMin, float xRangeMax, float yRangeMin, float yRangeMax) 
     {
+        if (xRangeMax == xRangeMin) {
```

Comment:
Alternatively, this change could apply only to type 3 functions if we created an override of the `interpolate` function in `PDFunctionType3`.

But avoiding divisions by 0 for other function types sounded sensible to me